blocks/library/block/index.js

```diff
@@ -0,0 +1,172 @@
+			attributes: null,
+		} );
+	}
+
+	setAttributes( attributes ) {
+		this.setState( ( prevState ) => ( {
+			attributes: { ...prevState.attributes, ...attributes },
+		} ) );
+	}
+
+	setName( name ) {
+		this.setState( { name } );
+	}
+
+	updateReusableBlock() {
+		const { name, attributes } = this.state;
+
+		// Use pickBy to include only changed (assigned) values in payload
+		const payload = pickBy( {
+			name,
+			attributes,
+		} );
+
+		this.props.updateReusableBlock( payload );
+		this.props.saveReusableBlock();
+		this.stopEditing();
+	}
+
+	render() {
+		const { focus, reusableBlock, convertBlockToStatic } = this.props;
+		const { isEditing, name, attributes } = this.state;
+
+		if ( ! reusableBlock ) {
+			return <Placeholder><Spinner /></Placeholder>;
+		}
+
+		const reusableBlockAttributes = { ...reusableBlock.attributes, ...attributes };
+		const blockType = getBlockType( reusableBlock.type );
+		const BlockEdit = blockType.edit || blockType.save;
+
+		// Generate a class name for the block's editable form
+		const generatedClassName = hasBlockSupport( blockType, 'className', true ) ?
+			getBlockDefaultClassname( reusableBlock.type ) :
+			null;
+		const className = classnames( generatedClassName, reusableBlockAttributes.className );
+
+		return [
+			// We fake the block being read-only by wrapping it with an element that has pointer-events: none
+			<div key="edit" style={ { pointerEvents: isEditing ? 'auto' : 'none' } }>
+				<BlockEdit
+					{ ...this.props }
+					focus={ isEditing ? focus : null }
+					attributes={ reusableBlockAttributes }
+					setAttributes={ isEditing ? this.setAttributes : noop }
+					className={ className }
+				/>
+			</div>,
+			focus && (
+				<ReusableBlockEditPanel
+					key="panel"
+					isEditing={ isEditing }
+					name={ name !== null ? name : reusableBlock.name }
+					isSaving={ reusableBlock.isSaving }
+					onEdit={ this.startEditing }
+					onDetach={ convertBlockToStatic }
+					onChangeName={ this.setName }
+					onSave={ this.updateReusableBlock }
+					onCancel={ this.stopEditing }
+				/>
+			),
+		];
+	}
+}
+
+const ConnectedReusableBlockEdit = connect(
+	( state, ownProps ) => ( {
+		reusableBlock: state.reusableBlocks.data[ ownProps.attributes.ref ],
+	} ),
+	( dispatch, ownProps ) => ( {
+		fetchReusableBlock() {
+			dispatch( {
+				type: 'FETCH_REUSABLE_BLOCKS',
+				id: ownProps.attributes.ref,
+			} );
+		},
+		updateReusableBlock( reusableBlock ) {
+			dispatch( {
+				type: 'UPDATE_REUSABLE_BLOCK',
+				id: ownProps.attributes.ref,
+				reusableBlock,
+			} );
+		},
+		saveReusableBlock() {
+			dispatch( {
+				type: 'SAVE_REUSABLE_BLOCK',
+				id: ownProps.attributes.ref,
+			} );
+		},
+		convertBlockToStatic() {
+			dispatch( {
+				type: 'CONVERT_BLOCK_TO_STATIC',
+				uid: ownProps.id,
+			} );
+		},
+	} )
+)( ReusableBlockEdit );
+
+registerBlockType( 'core/block', {
+	title: __( 'Reusable Block' ),
+	category: 'reusable-blocks',
+	isPrivate: true,
+
+	attributes: {
```

Comment:
I don't think you should need to define this on the client, since it should be bootstrapped from the server definition.

See #2529

Reply:
I'm hesitant to do this because removing the client-side definition means our test fixtures aren't able to test parsing a `<!-- wp:block ref="358b59ee-bab3-4d6f-8445-e8c6971a5605" /-->`.